lib/services/hooks-api-service.js

```diff
@@ -0,0 +1,91 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = hookApiServiceFactory;
+di.annotate(hookApiServiceFactory, new di.Provide('Http.Services.Api.Hooks'));
+di.annotate(hookApiServiceFactory,
+    new di.Inject(
+        'Services.Waterline',
+        'Logger',
+        '_',
+        'Errors'
+    )
+);
+function hookApiServiceFactory(
+    waterline,
+    Logger,
+    _,
+    Errors
+) {
+    var logger = Logger.initialize(hookApiServiceFactory);
+    function HooksApiService() {
+    }
+
+    /**
+     * Get hooks
+     * @return {Promise}
+     */
+    HooksApiService.prototype.getHooks = function (query) {
+        return waterline.hooks.find(query);
+    };
+
+    /**
+     * Create new hooks
+     * @param {Object} [req] HTTP request
+     * @param {Object} [res] HTTP response
+     * @return
+     */
+    HooksApiService.prototype.createHook = function(body) {
+        var self = this;
+        return waterline.hooks.findOne({url: body.url})
+        .then(function(entry){
+            if(_.isEmpty(entry)) {
+                return waterline.hooks.create(body)
+                .then(function(hook){
+                    return hook;
+                });
+            } else if (body.filters && !_.isEqual(body.filters, entry.filters)) {
+                var filters = entry.filters || [];
+                filters = _.uniq(filters.concat(body.filters));
+                entry.filters = filters;
+                return waterline.hooks.update(entry)
```

Comment:
`POST` is always to create a new resource, not update/modify the existing resource, so the logic about `hooks.update` should be removed.